src/NSwag.CodeGeneration.CSharp.Tests/ControllerGenerationFormatTests.cs

```diff
@@ -86,6 +87,82 @@ public void Bar()
             Assert.IsTrue(code.Contains("private ITestController _implementation;"));
             Assert.IsFalse(code.Contains("abstract class TestController"));
         }
+
+        [TestMethod]
+        public async Task When_partialcontrollermethodwithnorequestparameter_is_async_then_cancellationtoken_is_added()
+        {
+            //// Arrange
+            var swaggerGen = new WebApiToSwaggerGenerator(new WebApiToSwaggerGeneratorSettings());
+            var document = await swaggerGen.GenerateForControllerAsync<TestController>();
+
+            //// Act
+            var codeGen = new SwaggerToCSharpWebApiControllerGenerator(document, new SwaggerToCSharpWebApiControllerGeneratorSettings
+            {
+                ControllerStyle = CSharpControllerStyle.Partial
+            });
+            var code = codeGen.GenerateFile();
+
+            //// Assert
+            Assert.IsTrue(code.Contains("(System.Threading.CancellationToken cancellationToken)"));
+            Assert.IsTrue(code.Contains("_implementation.BarAsync(cancellationToken)"));
+            Assert.IsTrue(code.Contains("System.Threading.Tasks.Task BarAsync(System.Threading.CancellationToken cancellationToken)"));
+        }
+
+        [TestMethod]
+        public async Task When_partialcontrollermethodhasrequestparameter_is_async_then_cancellationtoken_is_added()
+        {
+            //// Arrange
+            var swaggerGen = new WebApiToSwaggerGenerator(new WebApiToSwaggerGeneratorSettings());
+            var document = await swaggerGen.GenerateForControllerAsync<TestController>();
+
+            //// Act
+            var codeGen = new SwaggerToCSharpWebApiControllerGenerator(document, new SwaggerToCSharpWebApiControllerGeneratorSettings
+            {
+            });
+            var code = codeGen.GenerateFile();
+
+            //// Assert
+            Assert.IsTrue(code.Contains("System.Threading.Tasks.Task<string> FooAsync(string test, bool test2, System.Threading.CancellationToken cancellationToken)"));
```

Comment:
Just a thought but the intent of the test is not so much "does it produce this exact code" but "does the code it produces do the right thing". It might make a better test to [compile the generated code](http://josephwoodward.co.uk/2016/12/in-memory-c-sharp-compilation-using-roslyn) and check you can call methods via reflexion.